app/src/main/java/io/heckel/ntfy/work/DeleteWorker.kt

```diff
@@ -59,6 +64,48 @@ class DeleteWorker(ctx: Context, params: WorkerParameters) : CoroutineWorker(ctx
         }
     }
 
+    private fun deleteExpiredIcons() {
+        Log.d(TAG, "Deleting icons for deleted notifications")
+        val resolver = applicationContext.contentResolver
+        val repository = Repository.getInstance(applicationContext)
+        val notifications = repository.getDeletedNotificationsWithIcons()
+        notifications.forEach { notification ->
+            try {
+                val icon = notification.icon ?: return
+                val contentUri = Uri.parse(icon.contentUri ?: return)
+                Log.d(TAG, "Deleting icon for notification ${notification.id}: ${icon.contentUri} (${icon.url})")
+                val deleted = resolver.delete(contentUri, null, null) > 0
+                if (!deleted) {
+                    Log.w(TAG, "Unable to delete icon for notification ${notification.id}")
+                }
+                val newIcon = icon.copy(
+                    contentUri = null,
+                )
+                val newNotification = notification.copy(icon = newIcon)
+                repository.updateNotification(newNotification)
+            } catch (e: Exception) {
+                Log.w(TAG, "Failed to delete icon for notification: ${e.message}", e)
+            }
+        }
+    }
```

Comment:
Correct me if I'm wrong, but now that we are sharing icon files across multiple notifications, deleting the icons for deleted notifications could delete the icon for non-deleted notifications. 

Assume this:
- notification 1, icon URL = https://example.com/icon.png, icon file = content://1234, deleted = true
- notification 2, icon URL = https://example.com/icon.png, icon file = content://1234, deleted = false

Now we run `deleteExpiredIcons`, which deletes `content://1234`, and notification 2 cannot display its icon anymore.

Given that you have `cleanIconCache`, would the solution not be to just delete this function?